### PR TITLE
feature: import token popup

### PR DESCRIPTION
**Pull request by ChriLnth**
## Changelog

- import token controller
- new popup screen

### Type of changes included:

- [x] Components
- [x] Business Logic
- [ ] Bug fixes
- [ ] Styling
- [ ] Code Refactor

### Clubhouse Tickets Related:

- [notion](https://www.notion.so/Add-importCustomToken-popup-to-plug-plug-inpage-provider-7ac68e4184b34854b774d2b687431dc1)

### Screenshots/Gifs:

https://www.loom.com/share/139b018965f94195bdc1dd7ab729b594

### Tested on:

- [x] Chrome
- [ ] Firefox
- [ ] Safari

### CODE REVIEW

File: source/Pages/Notification/components/ImportToken/styles.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    display: 'flex',
+    flexDirection: 'row',
+    justifyContent: 'flex-start',
+    padding: `${theme.spacing(1)}px 0`,
+  },
+  leftContainer: {
+    display: 'flex',
+    flexDirection: 'column',
+    justifyContent: 'space-evenly',
+  },
```
Comment (tomiir) [inline source/Pages/Notification/components/ImportToken/styles.js:32]: you could abstract this to `container` or something like that

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "plug",
-  "version": "0.5.2.2",
+  "version": "0.5.3.0",
```
Comment (tomiir) [inline package.json:3]: 0.5.3

File: source/Pages/Notification/components/ImportToken/index.jsx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+import PropTypes from 'prop-types';
+import { CONNECTION_STATUS } from '@shared/constants/connectionStatus';
+import { setAccountInfo } from '@redux/wallet';
+import { HANDLER_TYPES, sendMessage } from '@background/Keyring';
+import initConfig from '../../../../locales';
+import useStyles from './styles';
+import { reviewPendingTransaction } from '@modules/storageManager';
+import DisplayBox from '../Sign/components/Details/components/DisplayBox';
+
+i18n.use(initReactI18next).init(initConfig);
```
Comment (tomiir) [inline source/Pages/Notification/components/ImportToken/index.jsx:21]: why?

File: source/Pages/Notification/components/ImportToken/index.jsx [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+});
+
+portRPC.start();
+
+const ImportToken = ({ args, metadata, callId, portId, setOnTimeout, transactionId, }) => {
+  const classes = useStyles();
+  const { t } = useTranslation();
+  const dispatch = useDispatch();
+
+  const [loading, setLoading] = useState('');
```
Comment (tomiir) [inline source/Pages/Notification/components/ImportToken/index.jsx:36]: why use a string here? 
shouldn't it be a bool?

File: source/Pages/Notification/components/ImportToken/index.jsx [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+            />
+            <Button
+              variant="rainbow"
+              value={t('common.allow')}
+              onClick={() => handleImportToken(CONNECTION_STATUS.accepted)}
+              fullWidth
+              style={{ width: '96%' }}
+              loading={loading === CONNECTION_STATUS.accepted}
+              disabled={loading === CONNECTION_STATUS.refused}
+              wrapperStyle={{ textAlign: 'right' }}
```
Comment (tomiir) [inline source/Pages/Notification/components/ImportToken/index.jsx:109]: you could abstract most of these props to like `commonButtonProps` or something like that. 

Also 
`const isRefused = loading === CONNECTION_STATUS.refused;`
`const isAccepted = loading === CONNECTION_STATUS.accepted;`

**Review by tomiir — COMMENTED** [review]
Some minor improvements to the code but lgtm overall

File: source/Pages/Notification/components/ImportToken/index.jsx (hunk 4 above)
Comment (ChriLnth) [reply source/Pages/Notification/components/ImportToken/index.jsx:36]: probably, just to difference if its loading the approve or decline and only show the loading there

**Review by tomiir — APPROVED** [review]

**Pull request merged**